### DISCUSSION TITLE
How does the "cryptographically verifyable" part of crev work?

### DISCUSSION THREAD

**Discussion by arnohaase** [crev-dev/cargo-crev#589] [General]
I just came across crev, love the concepts, this looks great - thanks for the effort first of all!
I looked through the book but did not find anything about how cryptographic verification is done - the registry files referenced from registry.toml have neither signatures nor digests for the audited libraries.
Any pointer would be greatly appreciated.

**Comment by dpc** [comment]
If you mean the integrity of the package, then a deterministic checksum of package content is calculated and signed over in every review, so anyone can compare if the content of the package is the same as was reviewers reviewed. [Example](https://github.com/dpc/crev-proofs/blob/bc5a9b66a15e681b8c4ab24039062b1a34bfc31d/FYlr8YoYGVvDwHQxqEIs89reKKDy-oWisoO0qXXEfHE/reviews/2019-01-packages-Ua7DxQ.proof.crev#L12).


The checksum code is here: https://github.com/crev-dev/recursive-digest

**Reply by arnohaase** [reply]
excellent, I had missed that. Thanks!
